File: generate_inserts.js
const fs = require('fs');
const path = require('path');

function getTags(name, desc) {
    let tags = [];
    let text = (name + " " + desc).toLowerCase();
    
    const hasAny = (keywords) => keywords.some(k => text.includes(k));
    
    if (hasAny(["bánh gạo", "tokbokki", "topokki"])) tags.push("Bánh gạo/Tokbokki");
    if (hasAny(["mì", "mỳ", "udon", "chajang", "ramen", "miến"])) tags.push("Mì Hàn Quốc");
    if (hasAny(["xúc xích", "chả cá", "chả hải sản", "thanh cua", "surimi", "chả", "naruto", "chạo"])) tags.push("Xúc xích & Chả cá");
    if (hasAny(["gia vị", "xốt", "sốt", "tương", "dầu", "nước chấm", "nước mắm", "giấm", "muối", "đường", "bột ớt", "bột"])) tags.push("Gia vị & Xốt");
    if (text.includes("phô mai") || text.includes("mozzarella")) tags.push("Phô mai");
    if (hasAny(["viên thả lẩu", "đậu hủ", "cá viên", "tôm viên", "bò viên", "cua viên", "túi tiền"])) tags.push("Viên thả lẩu");
    if (text.includes("rong biển")) tags.push("Rong biển");
    if (hasAny(["gà rán", "đùi gà", "cánh gà", "gà chiên", "gà karaage", "gà viên"])) tags.push("Gà rán/Gà chiên");
    if (hasAny(["bánh sữa", "bánh xếp", "mandu", "há cảo", "xíu mại", "bánh bao", "chả giò", "bánh tuyết"])) tags.push("Dimsum & Bánh ăn vặt");
    
    if (tags.length === 0) tags.push("Khác");
    
    return [...new Set(tags)];
}

const filepath = path.join('c:\\', 'Users', 'LEGION', 'Desktop', 'web.env', 'san_pham.ini');
const content = fs.readFileSync(filepath, 'utf-8');
const lines = content.split('\n');

let products = [];
// Regex to match: number. [sku] name - Giá: price ₫ - Mô tả: desc[cite: 1]
const pattern = /^\d+\.\s+\[(.*?)\]\s+(.*?)\s+-\s+Giá:\s+(.*?)\s*₫?\s*-\s+Mô tả:\s+(.*?)(?:\[cite:.*?\])?$/;

for (let line of lines) {
    line = line.trim();
    if (!line || !/^\d+\./.test(line)) continue;
    
    const match = line.match(pattern);
    if (match) {
        const sku = match[1].trim();
        const name = match[2].trim();
        const price_str = match[3].trim();
        let desc = match[4].trim();
        
        // Remove trailing [cite: 1] if it wasn't caught
        desc = desc.replace(/\[cite:\s*\d+\]/g, '').trim();

        // Clean price
        const price_clean = price_str.replace(/[^\d]/g, '');
        const price = price_clean ? parseInt(price_clean, 10) : 0;
        
        const tags = getTags(name, desc);
        
        products.push({
            sku, name, price, description: desc, tags, image_url: null
        });
    }
}

const chunkSize = 50;
const chunks = [];
for (let i = 0; i < products.length; i += chunkSize) {
    chunks.push(products.slice(i, i + chunkSize));
}

const outputDir = path.join('c:\\', 'Users', 'LEGION', 'Desktop', 'web.env');

chunks.forEach((chunk, index) => {
    let sql = "INSERT INTO products (sku, name, price, description, tags, image_url) VALUES \n";
    let values = chunk.map(p => {
        const n = p.name.replace(/'/g, "''");
        const d = p.description.replace(/'/g, "''");
        const t_str = "ARRAY[" + p.tags.map(t => `'${t}'`).join(', ') + "]::TEXT[]";
        return `('${p.sku}', '${n}', ${p.price}, '${d}', ${t_str}, NULL)`;
    });
    
    sql += values.join(',\n') + ";\n";
    
    const filename = path.join(outputDir, `insert_chunk_${index + 1}.sql`);
    fs.writeFileSync(filename, sql, 'utf-8');
});

console.log(`Generated ${chunks.length} chunk files for ${products.length} products.`);

File: generate_inserts.py
import re
import os
import json

def get_tags(name, desc):
    tags = []
    text = (name + " " + desc).lower()
    if any(k in text for k in ["bánh gạo", "tokbokki", "topokki"]):
        tags.append("Bánh gạo/Tokbokki")
    if any(k in text for k in ["mì", "mỳ", "udon", "chajang", "ramen", "miến"]):
        tags.append("Mì Hàn Quốc")
    if any(k in text for k in ["xúc xích", "chả cá", "chả hải sản", "thanh cua", "surimi", "chả", "naruto", "chạo"]):
        tags.append("Xúc xích & Chả cá")
    if any(k in text for k in ["gia vị", "xốt", "sốt", "tương", "dầu", "nước chấm", "nước mắm", "giấm", "muối", "đường", "bột ớt", "bột"]):
        tags.append("Gia vị & Xốt")
    if "phô mai" in text or "mozzarella" in text:
        tags.append("Phô mai")
    if any(k in text for k in ["viên thả lẩu", "đậu hủ", "cá viên", "tôm viên", "bò viên", "cua viên", "túi tiền"]):
        tags.append("Viên thả lẩu")
    if "rong biển" in text:
        tags.append("Rong biển")
    if any(k in text for k in ["gà rán", "đùi gà", "cánh gà", "gà chiên", "gà karaage", "gà viên"]):
        tags.append("Gà rán/Gà chiên")
    if any(k in text for k in ["bánh sữa", "bánh xếp", "mandu", "há cảo", "xíu mại", "bánh bao", "chả giò", "bánh tuyết"]):
        tags.append("Dimsum & Bánh ăn vặt")
    if not tags:
        tags.append("Khác")
    return list(set(tags))

with open(r"c:\Users\LEGION\Desktop\web.env\san_pham.ini", "r", encoding="utf-8") as f:
    lines = f.readlines()

products = []
# Regex to match: number. [sku] name - Giá: price ₫ - Mô tả: desc[cite: 1]
pattern = re.compile(r'^\d+\.\s+\[(.*?)\]\s+(.*?)\s+-\s+Giá:\s+(.*?)\s*₫?\s*-\s+Mô tả:\s+(.*?)(?:\[cite:.*?\])?$')

for line in lines:
    line = line.strip()
    if not line or not re.match(r'^\d+\.', line):
        continue
    
    match = pattern.match(line)
    if match:
        sku = match.group(1).strip()
        name = match.group(2).strip()
        price_str = match.group(3).strip()
        desc = match.group(4).strip()
        
        # Clean price: remove dots, spaces, Liên hệ, etc.
        price_clean = re.sub(r'[^\d]', '', price_str)
        price = int(price_clean) if price_clean else 0
        
        tags = get_tags(name, desc)
        
        products.append({
            "sku": sku,
            "name": name,
            "price": price,
            "description": desc,
            "tags": tags,
            "image_url": None
        })

# Chunking into 50 products per file
chunk_size = 50
chunks = [products[i:i + chunk_size] for i in range(0, len(products), chunk_size)]

output_dir = r"c:\Users\LEGION\Desktop\web.env"

for i, chunk in enumerate(chunks):
    sql = "INSERT INTO products (sku, name, price, description, tags, image_url) VALUES \n"
    values = []
    for p in chunk:
        # Escape single quotes in strings
        n = p['name'].replace("'", "''")
        d = p['description'].replace("'", "''")
        t_str = "ARRAY[" + ", ".join([f"'{t}'" for t in p['tags']]) + "]::TEXT[]"
        values.append(f"('{p['sku']}', '{n}', {p['price']}, '{d}', {t_str}, NULL)")
    sql += ",\n".join(values) + ";"
    
    filename = os.path.join(output_dir, f"insert_chunk_{i+1}.sql")
    with open(filename, "w", encoding="utf-8") as f:
        f.write(sql)

print(f"Generated {len(chunks)} chunk files for {len(products)} products.")
